### ISSUE TITLE
$request->file(...) / $request->allFiles() is empty for Filepond fileuploads while using FormRequests

### ISSUE THREAD

**Issue by J87NL** [protonemedia/laravel-splade#126]
Hi Pascal,

Thanks for adding Filepond, I just finished my own custom Filepond upload implementation when #100 was added. So I refactored and I love it!

I use a FormRequest that ` implements HasSpladeFileUploads`, validate with `'filePath' => ['nullable',  'file'],`

My form contains:
`<x-splade-file name="filePath" :label="__('Bestand')" filepond accept="video/quicktime, video/mp4" server />`

I managed to get it working like this:
```
        // First I do an update on the model
        $model->update($request->safe()->except(['filePath']));

        // Then I add the file if one was uploaded
        if (!empty($request->validated('filePath'))) {
            $model->path = $request->validated('filePath')->store('files');
            $model->save();
        }
```

I have to do it this way because `$request->file(...)` and  `$request->allFiles()` are empty.
The snippet from the documentation (`$path = $request->file('filePath')->store('files'); `) is not working for this approach.

Is this expected behaviour or am I doing something wrong here?

**Comment by pascalbaljet** [comment]
Hmm, I can't reproduce the problem. I've copied the form element and validation rule, but it works. Could you set up a (demo) repository that I check out?

**Comment by J87NL** [comment]
Really strange, I made a new demo project for this issue and I can't reproduce either. 
Still in my project it doesn't work, so although all settings and versions seem to be the same, I must have messed up something. I let you know if I found out what it was!


Another small thing I came across though:

![afbeelding](https://user-images.githubusercontent.com/16107428/203614070-12703910-c249-44b7-b435-38a03cf59040.png)

Using `File::image()` gives this validation error:
![afbeelding](https://user-images.githubusercontent.com/16107428/203614389-3c96386a-4d39-4dcf-b510-bff91fd22300.png)

**Comment by pascalbaljet** [comment]
The `File::image()` error has been fixed in 1.0.1.

**Comment by J87NL** [comment]
After quite some debugging I found out what causes this behaviour, not sure why though:

```
<?php

namespace App\Http\Requests\Admin;

use Illuminate\Foundation\Http\FormRequest;
use Illuminate\Validation\Rule;
use ProtoneMedia\Splade\FileUploads\HasSpladeFileUploads;

class TestRequest extends FormRequest implements HasSpladeFileUploads
{
    public function rules(): array
    {
        return [
            'title' => [
                'required',
                'string',
                'max:255',

                // Adding the unique-rule below results in $request->hasFile('filePath') == false
                // and $request->allFiles() / $request->file('filePath') to be empty
                Rule::unique('videos')->ignore($this->video->id ?? null)->whereNull('deleted_at'),
            ],

            // ... some more rules

            'filePath' => [
                'nullable',
                'file',
            ],
        ];
    }
}
```

`$request->validated('filePath')->store('files');` is still working fine though.

I'm not quite sure if this is a Splade-issue, what do you think Pascal?

**Comment by pascalbaljet** [comment]
I've found the problem! When you call `$this->video` in the `ignore` method, under the hood, it calls the `__get()` method of the `Request` class, which eventually calls the `allFiles()` method. The validation rules are evaluated *before* Splade converts the temporary FilePond upload to a `UploadedFile` instance, but the `allFiles()` method already caches the files in a `$convertedFiles` property:

https://github.com/laravel/framework/blob/99aede05b96acda561417511f8350188d45994f5/src/Illuminate/Http/Concerns/InteractsWithInput.php#L523

Long story short, Splade now clears the `$convertedFiles` property after it has converted the FilePond upload :)

Fixed in v1.0.4.

**Comment by J87NL** [comment]
Awesome Pascal, thanks and keep up the good work!
